internal/utils/misc.go

```diff
@@ -376,6 +381,80 @@ func InstallOrUpgradeDeno(ctx context.Context, fsys afero.Fs) error {
 	return nil
 }
 
+func isBuildScriptModified(fsys afero.Fs, buildScriptPath string) (bool, error) {
+	bs, err := afero.ReadFile(fsys, buildScriptPath)
+	if err != nil {
+		if errors.Is(err, fs.ErrNotExist) {
+			return true, nil
+		}
+		return false, err
+	}
+
+	es, err := fs.ReadFile(eszipEmbedDir, "eszip/build.ts")
+	if err != nil {
+		return false, err
+	}
+
+	// compare the md5 checksum of current build script with user's copy.
+	// if the checksums doesn't match, build script is modified.
+	return md5.Sum(bs) != md5.Sum(es), nil
+}
+
+// Copy ESZIP scripts needed for function deploy, returning the build script path or an error.
+func CopyEszipScripts(ctx context.Context, fsys afero.Fs) (string, error) {
+	home, err := os.UserHomeDir()
+	if err != nil {
+		return "", err
+	}
+
+	supabasePath := filepath.Join(home, ".supabase")
+	scriptDirPath := filepath.Join(supabasePath, "eszip")
+	buildScriptPath := filepath.Join(scriptDirPath, "build.ts")
```

Comment:
nitpick: can we make this path relative to `GetDenoPath`? slightly easier to keep track

Reply:
Good idea! fixed in 2189242